lib/rdoc/generator/template/snapper/class.rhtml

```diff
@@ -1,163 +1,113 @@
-              <%- if method.token_stream then -%>
-              <span class="method-click-advice">click to toggle source</span>
+
+        <%- unless constants.empty? -%>
+          <section>
+            <h3 class="section-title">Constants<hr/></h3>
+
+            <dl>
+              <%- constants.each do |const| -%>
+                <dt><%= const.name %>
+                <dd><%= const.comment ? const.description.strip : "(Not documented)" %></dd>
               <%- end -%>
-            </div>
-            <%- end -%>
-          </div>
-
-          <div class="method-description">
-            <%- if method.comment then -%>
-            <%= method.description.strip %>
-            <%- else -%>
-            <p class="missing-docs">(Not documented)
-            <%- end -%>
-            <%- if method.calls_super then -%>
-              <div class="method-calls-super">
-                Calls superclass method
-                <%=
-                    method.superclass_method ?
-                    method.formatter.link(method.superclass_method.full_name, method.superclass_method.full_name) : nil
-                %>
+            </dl>
+          </section>
+        <%- end -%>
+
+        <% unless attributes.empty? %>
+          <section>
+            <h3 class="section-title">Attributes<hr/></h3>
+
+            <% attributes.each do |attrib| %>
+              <div>
+                <h4 class="method-name">
+                  <%= h attrib.name %> <span class="attribute-type">(<%= h attrib.definition %>)</span>
+                </h4>
+
+                <div class="description">
+                  <%= attrib.comment ? attrib.description.strip : "(Not document)" %>
+                </div>
               </div>
-            <%- end -%>
-
-            <%- if method.token_stream then -%>
-            <div class="method-source-code" id="<%= method.html_name %>-source">
-              <pre><%= method.markup_code %></pre>
-            </div>
-            <%- end -%>
-          </div>
-
-          <%- unless method.aliases.empty? then -%>
-          <div class="aliases">
-            Also aliased as: <%= method.aliases.map do |aka|
-              if aka.parent then # HACK lib/rexml/encodings
-                %{<a href="#{klass.aref_to aka.path}">#{h aka.name}</a>}
-              else
-                h aka.name
-              end
-            end.join ", " %>
-          </div>
-          <%- end -%>
-
-          <%- if method.is_alias_for then -%>
-          <div class="aliases">
-            Alias for: <a href="<%= klass.aref_to method.is_alias_for.path %>"><%= h method.is_alias_for.name %></a>
-          </div>
-          <%- end -%>
-        </div>
-
-      <%- end -%>
+            <% end %>
+          </section>
+        <% end %>
+
+        <% klass.methods_by_type(section).select { |_type, v| v.any? }.each do |type, visibilities|
+          visibilities.each do |visibility, methods| %>
+            <% next if methods.empty? %>
+
+            <section>
```

Comment:
I love the use of sections. Maybe we could give them a top margin so things are a bit less cramped, like this:

![Screenshot 2023-07-28 at 3 57 51 PM](https://github.com/Shopify/rdoc/assets/6955296/2025a2a3-5495-467a-82fe-73544eee0b7d)

Something like:

``` css
section {
  margin-top: 50px;
}
```

Or even better:

``` css
section {
  margin-top: calc(var(--space) * 3);
}
```

Reply:
Great suggestion!